### PR TITLE
Avatar priority (hero) zone improvements

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Ticket: https://highfidelity.manuscript.com/f/cases/21154/

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17771/HighFidelity-Beta-PR15067-5a8ecdf.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17771/HighFidelity-Quest-Beta-PR15067-5a8ecdf.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17771/HighFidelity-Beta-PR15067-5a8ecdf.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17771/HighFidelity-Beta-Interface-PR15067-5a8ecdf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17771/HighFidelity-Beta-PR15067-5a8ecdf.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17771/HighFidelity-Beta-Interface-PR15067-5a8ecdf.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17775/HighFidelity-Beta-PR15067-64a2025.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17775/HighFidelity-Quest-Beta-PR15067-64a2025.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17775/HighFidelity-Beta-PR15067-64a2025.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17775/HighFidelity-Beta-Interface-PR15067-64a2025.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17775/HighFidelity-Beta-PR15067-64a2025.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17775/HighFidelity-Beta-Interface-PR15067-64a2025.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17785/HighFidelity-Beta-PR15067-d391d4a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17785/HighFidelity-Quest-Beta-PR15067-d391d4a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17785/HighFidelity-Beta-PR15067-d391d4a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17785/HighFidelity-Beta-Interface-PR15067-d391d4a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17785/HighFidelity-Beta-PR15067-d391d4a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17785/HighFidelity-Beta-Interface-PR15067-d391d4a.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17804/HighFidelity-Beta-PR15067-3cc932d.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17804/HighFidelity-Quest-Beta-PR15067-3cc932d.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17804/HighFidelity-Beta-PR15067-3cc932d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17804/HighFidelity-Beta-Interface-PR15067-3cc932d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17804/HighFidelity-Beta-PR15067-3cc932d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17804/HighFidelity-Beta-Interface-PR15067-3cc932d.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17832/HighFidelity-Beta-PR15067-83c9381.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17832/HighFidelity-Quest-Beta-PR15067-83c9381.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17832/HighFidelity-Beta-PR15067-83c9381.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17832/HighFidelity-Beta-Interface-PR15067-83c9381.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17832/HighFidelity-Beta-PR15067-83c9381.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17832/HighFidelity-Beta-Interface-PR15067-83c9381.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17854/HighFidelity-Beta-PR15067-122b62a.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17854/HighFidelity-Quest-Beta-PR15067-122b62a.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17854/HighFidelity-Beta-PR15067-122b62a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17854/HighFidelity-Beta-Interface-PR15067-122b62a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17854/HighFidelity-Beta-PR15067-122b62a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17854/HighFidelity-Beta-Interface-PR15067-122b62a.exe

File: assignment-client/src/avatars/AvatarMixerSlave.cpp [hunk 1]
```diff
@@ -353,8 +353,11 @@ void AvatarMixerSlave::broadcastAvatarDataToAgent(const SharedNodePointer& node)
 
     // When this is true, the AvatarMixer will send Avatar data to a client
     // about avatars they've ignored or that are out of view
-    bool PALIsOpen = nodeData->getRequestsDomainListData();
-    bool PALWasOpen = nodeData->getPrevRequestsDomainListData();
+    bool PALIsOpen = destinationNodeData->getRequestsDomainListData();
+    bool PALWasOpen = destinationNodeData->getPrevRequestsDomainListData();
+    if (PALIsOpen) {
+        qCWarning(avatars) << "PAL is open:" << avatar.getSessionDisplayName();
```
Comment (Atlante45) [inline assignment-client/src/avatars/AvatarMixerSlave.cpp]: Potential log spam

File: assignment-client/src/avatars/AvatarMixerSlave.cpp [hunk 2]
```diff
@@ -521,6 +526,9 @@ void AvatarMixerSlave::broadcastAvatarDataToAgent(const SharedNodePointer& node)
                     _stats.overBudgetAvatars++;
                     detail = AvatarData::PALMinimum;
                 } else {
+                    if (currentVariant == kHero) {
+                        qCWarning(avatars) << "Overbudget break with hero avatars!" << destinationNode->getUUID().toString();
```
Comment (Atlante45) [inline assignment-client/src/avatars/AvatarMixerSlave.cpp]: Potential log spam

File: assignment-client/src/avatars/MixerAvatar.h [hunk 3]
```diff
@@ -19,11 +19,11 @@
...
 class MixerAvatar : public AvatarData {
 public:
-    bool getPriorityAvatar() const { return  _bPriorityAvatar; }
-    void setPriorityAvatar(bool bPriorityAvatar) { _bPriorityAvatar = bPriorityAvatar; }
+    bool getHasPriority() const { return  _bHasPriority; }
+    void setHasPriority(bool bPriorityAvatar) { _bHasPriority = bPriorityAvatar; }
 
 private:
-    bool _bPriorityAvatar { false };
+    bool _bHasPriority { false };
```
Comment (Atlante45) [inline assignment-client/src/avatars/MixerAvatar.h]: Naming standard, should be `_hasPriority`, `hasPriority`

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17858/HighFidelity-Beta-PR15067-743d1a5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17858/HighFidelity-Beta-Interface-PR15067-743d1a5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17858/HighFidelity-Beta-PR15067-743d1a5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17858/HighFidelity-Quest-Beta-PR15067-743d1a5.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17858/HighFidelity-Beta-PR15067-743d1a5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17858/HighFidelity-Beta-Interface-PR15067-743d1a5.exe

**Pull request merged**